src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.IProduceBlockTransactionsStrategy.cs

```diff
@@ -0,0 +1,63 @@
+﻿//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core;
+using Nethermind.Core.Collections;
+using Nethermind.Core.Specs;
+using Nethermind.Evm;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.State;
+using Nethermind.State.Proofs;
+using Nethermind.TxPool;
+using Nethermind.TxPool.Comparison;
+
+namespace Nethermind.Blockchain.Processing
+{
+    public partial class BlockProcessor
+    {
+        protected interface IProduceBlockTransactionsStrategy : IBlockProcessor.IBlockTransactionsStrategy
```

Comment:
strange naming
maybe:
IBlockProcessingMode
ProducerBlockProcessingMode
VerifierBlockProcessingMode

Reply:
maybe mode not the best because it is not enum

Reply:
keep thinking, something is off

Reply:
We changed the name 3 times already, don't have a good one yet :(